Docs/source/install/hpc/cori.rst

```diff
@@ -45,25 +45,25 @@ And install ADIOS2, BLAS++ and LAPACK++:
    # c-blosc (I/O compression)
    git clone -b v1.21.1 https://github.com/Blosc/c-blosc.git src/c-blosc
    rm -rf src/c-blosc-knl-build
-   cmake -S src/c-blosc -B src/c-blosc-knl-build -DBUILD_TESTS=OFF -DBUILD_BENCHMARKS=OFF -DDEACTIVATE_AVX2=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/c-blosc-1.12.1-knl-install
+   cmake -S src/c-blosc -B src/c-blosc-knl-build -DBUILD_TESTS=OFF -DBUILD_BENCHMARKS=OFF -DDEACTIVATE_AVX2=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/knl/c-blosc-1.12.1-install
    cmake --build src/c-blosc-knl-build --target install --parallel 16
 
    # ADIOS2
    git clone -b v2.7.1 https://github.com/ornladios/ADIOS2.git src/adios2
    rm -rf src/adios2-knl-build
-   cmake -S src/adios2 -B src/adios2-knl-build -DADIOS2_USE_Blosc=ON -DADIOS2_USE_Fortran=OFF -DADIOS2_USE_Python=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/adios2-2.7.1-knl-install
+   cmake -S src/adios2 -B src/adios2-knl-build -DADIOS2_USE_Blosc=ON -DADIOS2_USE_Fortran=OFF -DADIOS2_USE_Python=OFF -DCMAKE_INSTALL_PREFIX=$HOME/sw/knl/adios2-2.7.1-install
    cmake --build src/adios2-knl-build --target install --parallel 16
```

Comment:
I get errors:
[ 28%] Building CXX object source/adios2/CMakeFiles/adios2_core.dir/engine/dataman/DataManMonitor.cpp.o
In file included from /global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.cpp:17:
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.h:34:20: error: 'std::shared_ptr' has not been declared
   34 |     void Send(std::shared_ptr<std::vector<char>> buffer);
      |                    ^~~~~~~~~~
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.h:34:30: error: expected ',' or ...' before '<' token
   34 |     void Send(std::shared_ptr<std::vector<char>> buffer);
      |                              ^
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.h:35:10: error: 'shared_ptr' in namespace 'std' does not name a template type
   35 |     std::shared_ptr<std::vector<char>> Receive();
      |          ^~~~~~~~~~
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.h:17:1: note: 'std::shared_ptr' is defined in header '<memory>'; did you forget to '#include <memory>'?
   16 | #include <thread>
  +++ |+#include <memory>
   17 | 
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.cpp:85:6: error: variable or field 'Send' declared void
   85 | void ZmqPubSub::Send(std::shared_ptr<std::vector<char>> buffer)
      |      ^~~~~~~~~
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.cpp:85:27: error: 'shared_ptr' is not a member of 'std'
   85 | void ZmqPubSub::Send(std::shared_ptr<std::vector<char>> buffer)
      |                           ^~~~~~~~~~
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.cpp:18:1: note: 'std::shared_ptr' is defined in header '<memory>'; did you forget to '#include <memory>'?
   17 | #include "ZmqPubSub.h"
  +++ |+#include <memory>
   18 | 
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.cpp:85:54: error: expected primary-expression before '>' token
   85 | void ZmqPubSub::Send(std::shared_ptr<std::vector<char>> buffer)
      |                                                      ^~
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.cpp:85:57: error: 'buffer' was not declared in this scope; did you mean 'setbuffer'?
   85 | void ZmqPubSub::Send(std::shared_ptr<std::vector<char>> buffer)
      |                                                         ^~~~~~
      |                                                         setbuffer
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.cpp:93:6: error: 'shared_ptr' in namespace 'std' does not name a template type
   93 | std::shared_ptr<std::vector<char>> ZmqPubSub::Receive()
      |      ^~~~~~~~~~
/global/homes/r/rsand/src/adios2/source/adios2/toolkit/zmq/zmqpubsub/ZmqPubSub.cpp:93:1: note: 'std::shared_ptr' is defined in header '<memory>'; did you forget to '#include <memory>'?
   93 | std::shared_ptr<std::vector<char>> ZmqPubSub::Receive()
      | ^~~
[ 28%] Building CXX object source/adios2/CMakeFiles/adios2_core.dir/engine/dataman/DataManReader.cpp.o
gmake[2]: *** [source/adios2/CMakeFiles/adios2_core.dir/build.make:1084: source/adios2/CMakeFiles/adios2_core.dir/toolkit/zmq/zmqpubsub/ZmqPubSub.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[1]: *** [CMakeFiles/Makefile2:3506: source/adios2/CMakeFiles/adios2_core.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2

Reply:
Haven't checked patch yet, but `-DADIOS2_USE_ZeroMQ=OFF` in the configure flag leads to successful build